### PR TITLE
Create approve modal

### PR DESCRIPTION
**Pull request by BenjaminParment**

### CODE REVIEW

File: src/components/reusables/list/row-definitions/approveTransactionRowDef.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+        key: "Product",
+        extractContent: (data: TransactionRow) => data.product,
+    },
+    {
+        key: "Transaction date",
+        extractContent: (data: TransactionRow) => formatDate(data.createdAt),
+    },
+    {
+        key: "Amount",
+        extractContent: (data: TransactionRow) => `£${data.amount}`,
```
Comment (floyd924) [inline src/components/reusables/list/row-definitions/approveTransactionRowDef.ts]: Should there be two different currencies here? £$

File: src/components/reusables/modal/approve-modal/ApproveModal.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    private onCancel = (): void => {
+        //Redirect to transaction page
+    };
+
+    private onApprove = (): void => {
+        // Approve action
+    };
+
+    render = () => (
+        <Modal isVisible={this.props.isVisible} title="Approve transaction" hideModal={this.props.closeModal}>
```
Comment (floyd924) [inline src/components/reusables/modal/approve-modal/ApproveModal.tsx]: do we need to provide a default value for `isVisible` or is it always handled by the parent?

File: src/components/reusables/list/row-definitions/approveTransactionRowDef.ts (hunk 1 above)
Comment (BenjaminParment) [reply src/components/reusables/list/row-definitions/approveTransactionRowDef.ts]: nah the dollar sign is just used to use data.amount in string

File: src/components/reusables/modal/approve-modal/ApproveModal.tsx (hunk 2 above)
Comment (BenjaminParment) [reply src/components/reusables/modal/approve-modal/ApproveModal.tsx]: yeah, the isVisible is handled by Modal which is a generic

**Pull request merged**